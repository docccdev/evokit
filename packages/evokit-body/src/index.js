import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
