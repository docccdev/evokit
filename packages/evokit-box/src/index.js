// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
