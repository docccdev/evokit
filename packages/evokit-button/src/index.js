import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
