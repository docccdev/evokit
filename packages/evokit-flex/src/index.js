import { createBlock } from 'evokit';

export const Flex = createBlock('div', 'flex', [
    'direction',
    'display',
    'items',
    'wrap',
    ['content', [
        'content-align',
        'content-justify',
    ]],
]);

export const FlexItem = createBlock('div', 'flex__item', [
    'align',
    'display',
    'order',
]);

Flex.Item = FlexItem;
