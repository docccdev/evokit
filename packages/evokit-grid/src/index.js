import { createBlock } from 'evokit';

export const Grid = createBlock('div', 'grid', [
    'align',
    'column',
    'direction',
    'display',
    'divider',
    'divider-column',
    'height',
    'valign',
    'wrap',
    ['divider-indent', [
        'divider-indent-x',
        'divider-indent-y',
    ]],
    ['indent', [
        'indent-x',
        'indent-y',
    ]],
]);

export const GridItem = createBlock('div', 'grid__item', [
    'display',
    'order',
    'width',
]);

Grid.Item = GridItem;
