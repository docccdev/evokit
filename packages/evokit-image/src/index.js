import { createBlock } from 'evokit';

export const Image = createBlock('img', 'image', [
    'align',
    'alt',
    'border',
    'border-bottom',
    'border-left',
    'border-lr',
    'border-right',
    'border-tb',
    'border-top',
    'display',
    'fit',
    'mirror',
    'valign',
    'width',
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
]);
