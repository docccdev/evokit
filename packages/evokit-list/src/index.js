import { createBlock } from 'evokit';

export const List = createBlock('ul', 'list', [
    'color',
    'display',
    'divider',
    'divider-indent',
    'indent',
    'style',
]);

export const ListItem = createBlock('li', 'list__item', ['display']);

List.Item = ListItem;
