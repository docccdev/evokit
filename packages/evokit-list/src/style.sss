@import './mixins.sss'

@custom-media --ek-list-media-small only screen and (min-width: 480px)
@custom-media --ek-list-media-medium only screen and (min-width: 768px)
@custom-media --ek-list-media-large only screen and (min-width: 960px)
@custom-media --ek-list-media-wide only screen and (min-width: 1200px)
@custom-media --ek-list-media-huge only screen and (min-width: 1400px)

:root
    // prop 'list-indent'
    --ek-list-indent-xxs: 5px
    --ek-list-indent-xs: 10px
    --ek-list-indent-s: 15px
    --ek-list-indent-m: 20px
    --ek-list-indent-l: 25px
    --ek-list-indent-xl: 30px
    --ek-list-indent-xxl: 35px
    --ek-list-indent-3xl: 40px
    --ek-list-indent-4xl: 45px
    --ek-list-indent-5xl: 50px

.list
    display: block
    list-style: none
    padding: 0
    margin: 0

.list__item
    display: block

/**
 * Display:
 *
 *  - list_display_block
 *  - list_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .list_display_$(value)
        display: $value

/**
 * Item display:
 *
 *  - list__item_display_block
 *  - list__item_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .list__item_display_$(value)
        display: $value

/**
 * Indent:
 *
 *  - list_indent_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, 3xl, 4xl, 5xl)
    @mixin media .list_indent_$(value)
        @if $value == none
            margin-top: 0
            margin-bottom: 0

            & > .list__item
                padding-top: 0
                padding-bottom: 0
        @else
            margin-top: calc(var(--ek-list-indent-$(value)) / 2  * -1)
            margin-bottom: calc(var(--ek-list-indent-$(value)) / 2  * -1)

            & > .list__item
                padding-top: calc(var(--ek-list-indent-$(value)) / 2)
                padding-bottom: calc(var(--ek-list-indent-$(value)) / 2)

/**
 * Divider indent:
 *
 *  - list_divider-indent_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, 3xl, 4xl, 5xl)
    @mixin media .list_divider-indent_$(value)
        @if $value == none
            margin-top: 0
            margin-bottom: 0

            & > .list__item
                padding-top: 0
                padding-bottom: 0
        @else
            margin-top: calc(var(--ek-list-indent-$(value)) * -1)
            margin-bottom: calc(var(--ek-list-indent-$(value)) * -1)

            & > .list__item
                padding-top: var(--ek-list-indent-$(value))
                padding-bottom: var(--ek-list-indent-$(value))

/**
 * Style:
 *
 *  - list_style_*
 *
 **/

@each $value in (dash, decimal, disc)
    .list_style_$(value)
        @if $value == decimal
            counter-reset: item-index

        & > .list__item
            padding-left: 20px
            position: relative

            &:before
                position: absolute
                margin-left: -20px
                width: 20px
                text-align: center
                @if $value == dash
                    content: '\2014'
                @else if $value == decimal
                    content: counter(item-index) '.'
                    counter-increment: item-index
                @else if $value == disc
                    content: '\2022'
                    font-weight: bold
