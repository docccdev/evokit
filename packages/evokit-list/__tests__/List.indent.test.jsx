import React from 'react';
import { shallow } from 'enzyme';
import { List } from '../src';

describe('<List list-indent="..." />', () => {
    it('props list-indent', () => {
        const wrapper = shallow(<List list-indent='value' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<ul class="ek-list ek-list_indent_value"></ul>');
    });
});
