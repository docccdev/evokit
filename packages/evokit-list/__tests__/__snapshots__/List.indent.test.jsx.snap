// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List list-indent="..." /> props list-indent 1`] = `
<ul
  className="ek-list ek-list_indent_value"
/>
`;
